peachjam/models/user_following.py

```diff
@@ -201,110 +204,40 @@ def get_new_followed_documents(self):
         )
         if self.last_alerted_at:
             qs = qs.filter(created_at__gt=self.last_alerted_at)
-
-        return {
-            "followed_object": self.followed_object,
-            "documents": qs[:10],
-        }
+        return qs[:10]
 
     @classmethod
-    def update_and_alert(cls, user):
-        follows = UserFollowing.objects.filter(user=user)
+    def update_timeline(cls, user):
+        follows = cls.objects.filter(user=user)
         log.info(f"Found {follows.count()} follows for user {user.pk}")
-        followed_documents = []
         for follow in follows:
             new = follow.get_new_followed_documents()
-            if new["documents"]:
+            if new:
                 log.info(
-                    f"Found {new['documents'].count()} new documents for {new['followed_object']}"
+                    f"Found {new.count()} new documents for {follow.followed_object}"
                 )
+                today = timezone.now().date()
+                timeline_event = TimelineEvent.objects.filter(
```

Comment:
use objects.get_or_create to avoid race conditions.

Reply:
done